charts/pega/charts/installer/templates/_pega-installer-job.tpl

```diff
@@ -5,6 +5,13 @@ apiVersion: batch/v1
 metadata:
   name: {{ .name }}
   namespace: {{ .root.Release.Namespace }}
+{{- if and .root.Values.waitForJobCompletion (or (eq .root.Values.global.actions.execute "install") (eq .root.Values.global.actions.execute "upgrade")) }}
+  annotations:
+    # Forces Helm to wait for the install or upgrade to complete.
+    "helm.sh/hook": post-install
+    "helm.sh/hook-weight": "0"
+    "helm.sh/hook-delete-policy": before-hook-creation
```

Comment:
How was the delete-policy chosen?  Would hook-succeeded make more sense?  (I'm not sure I fully understand how this policy is used)

Reply:
There are three hook policies.

From https://helm.sh/docs/topics/charts_hooks/:
Annotation Value | Description
-- | --
before-hook-creation | Delete the previous resource before a new hook is launched (default)
hook-succeeded | Delete the resource after the hook is successfully executed
hook-failed | Delete the resource if the hook failed during execution

Originally we had hook-succeeded, but then it was brought up that unless the user has a log manager to push install logs away from Kubernetes, these would be lost immediately when the install finishes. We figure it is safer to keep the job around so people can fetch those logs as they please, as we often want to keep them even on success (especially upgrade logs). This also more closely emulates existing behavior.